app/furniture/marketplace/archivable.rb

```diff
@@ -0,0 +1,30 @@
+class Marketplace
+  module Archivable
+    def self.included(model)
+      model.include(Discard::Model)
+      model.alias_method :archive, :discard
+      model.alias_method :archived?, :discarded?
+      model.alias_method :available?, :kept?
+      model.alias_method :retrieve, :undiscard
+
+      model.scope(:archived, -> { discarded })
+      model.scope(:available, -> { kept })
```

Comment:
I like having all of this in a module a lot! Expecting models within the Marketplace to be able to answer `with_orders?` also makes a lot of sense.

My one nit is that I would prefer `not_archived` to `available`, because it seems clearer. `Available` makes me think of "product not available in this delivery area" or "product not inventory". But I won't fight you on the name at this point. 😺

Reply:
I was having similar feels re: Available; I'm down for `unarchived`!